### ISSUE TITLE
Split Squiz.PHP.DisallowMultipleAssignments.Found into two different error codes

### ISSUE THREAD

**Issue by evilebottnawi** [squizlabs/PHP_CodeSniffer#2083]
Code:
```php
if ($app->runningInConsole() && ($input = new ArgvInput())->hasParameterOption('--env')) {
}
```

**Comment by gsherwood** [comment]
The error message is `Assignments must be the first block of code on a line`. The sniff doesn't like the  fact that you've used an assignment inside a condition. It wants assignments to be the only statement on a line. This also bans things like `$foo = $bar = true;`.

**Comment by evilebottnawi** [comment]
@gsherwood based on name of sniff i wait what it is only check multiple assignments, but in my example no multiple assignments. Name rule is misleading. For me it is false positive, we already exclude `=` in function and method and i think we should also exclude check from `if/while/etc`

**Comment by evilebottnawi** [comment]
Also PSR-12 contain this sniff https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/PSR12/ruleset.xml, but PSR-12 do not forbid it to use

**Comment by gsherwood** [comment]
> we already exclude = in functions and methods

They are excluded in functions and methods because they are the only way to specify a default value for a function argument. When used in if/while/etc they are being used to do assignments, which are self-contained statements. The if/while/etc is already a statement, hence multiple statements.

**Comment by evilebottnawi** [comment]
@gsherwood maybe be can add option for this?

**Comment by gsherwood** [comment]
> maybe be can add option for this?

Actually, I think maybe adding a second error message code could work as well.

So you'd have the existing Squiz.PHP.DisallowMultipleAssignments.Found and Squiz.PHP.DisallowMultipleAssignments.FoundInControlStructure (or something like that). That way, you can mute the FoundInControlStructure error in your ruleset if you don't want it. Adding a sniff code for a sniff that only produces one error message is also backwards compatible, so it should work for all existing rulesets.

Does that sound like a good idea to you?

**Comment by evilebottnawi** [comment]
@gsherwood good idea :+1:

**Comment by paulschreiber** [comment]
I ran in to this as well. See #1679.

**Comment by jrfnl** [comment]
PR #2402 addresses this issue.
